Interni/NormeDiProgetto/res/sections/processi_primari/processi_primari.tex

```diff
@@ -1,2 +1,237 @@
+\item \textbf{Design pattern}: devono essere descritti i design pattern utilizzati per realizzare l'architettura corredati da una descrizione, anche grafica, che ne descriva il significato e la struttura;
+\item \textbf{Tracciamento delle componenti}: ogni requisito deve riferirsi al componente che lo soddisfa;
+\item \textbf{Test di integrazione}: l'unione delle parti permette di verificare che ogni componente del sistema funzioni nella maniera voluta.
+\item \textbf{Product baseline}: Redatta dal progettista, dovrà includere:
+    \begin{itemize}
+        \item \textbf{definizione delle classi}: ogni classe dovrà essere descritta in modo breve e coinciso;
+        \item \textbf{tracciamento delle classi}: ogni requisito deve essere tracciato in modo che per ognuno esista una classe che lo soddisfi;
+        \item \textbf{test di unità}: devono essere definiti al fine di verificare che le parti funzionino individualmente nel modo stabilito.
+        \end{itemize}
+\end{itemize}
+
+\subsubsection{Codifica}
+\paragraph{Scopo}
+Questa attività ha lo scopo di normare l'effettiva realizzazione del prodotto software.  In questa fase si concretizza attraverso la programmazione quanto è stato progettato.
+
+\paragraph{Descrizione}
+La scrittura del codice dovrà perseguire gli obiettivi di qualità definiti nel Piano di Qualifica.
+
+\paragraph{Aspettative}
+Obiettivo dell'attività è la creazione del prodotto software conforme alle aspettative e richieste del proponente.  L'adozione di regole specifiche in questa fase è fondamentale per perseguire gli obiettivi di qualità e poter agevolare le successive fasi di manutenzione.
+
+\paragraph{Stile di Codifica}
+Ciascun programmatore è tenuto a rispettare le seguenti norme:
+
+\begin{itemize}
+\item \textbf{Indentazione}: i blocchi innestati devono essere correttamente indentati, usando per ciascun livello di indentazione quattro (4) spazi (fanno eccezione i commenti).  Tale norma può essere facilmente perseguita mediante un' opportuna configurazione dell'editor di codice;
+\item \textbf{Parentesizzazione}: si stabilisce che le parentisi di delimitazione dei costrutti vanno inserite in linea e non al di sotto di essi;
+\item \textbf{Scrittura dei metodi}: se possibile è auspicabile che tutti i metodi rispettino le seguenti norme ove possibile:
+\begin{itemize}
+\item prima lettera del nome minuscolo utilizzando una notazione "camel case" per le successive parole;
+\item tra il nome del metodo e l'eventuale parentesi di apertura deve essere inserita una singola spaziatura.
+\end{itemize}
+\item \textbf{Classi}: i nomi delle classi devono iniziare sempre con una lettera maiuscola;
+\item \textbf{Costanti}: i nome delle costanti devono essere tutti in maiuscolo ;
+\item \textbf{Lingua}: i nomi di variabili, costruttori, metodi, classi e commenti vanno scritti in lingua inglese.
+\end{itemize}
+
+
+
+\subsubsection{Metriche}
+\paragraph{tipi di metrica che andremo ad usare}
+
```

Comment:
qui vanno inseriti controlli di analisi statica, come complessità ciclomatica, lunghezza massima delle righe di codice, complessità massima delle espressioni booleane...

Reply:
Si il problema è che ancora non sappiamo cosa andremo a fare. Infatti RRR non citano minimamente queste cose nella prima versione delle NDP in processi primari. Propongo di toglierle